ivre/db/mongo.py

```diff
@@ -3376,6 +3387,32 @@ def update(self, spec, **kargs):
 """
         self.db[self.colname_passive].update(spec, {'$set': kargs})
 
+    @staticmethod
+    def _fix_values_size(spec):
+        # Finally we prepare the record to be stored. For that, we make
+        # sure that no indexed value has a size greater than MAXVALLEN. If
+        # so, we replace the value with its SHA1 hash and store the
+        # original value in full[original column name].
+        for key in ['value', 'targetval']:
+            if len(spec.get(key, "")) >  utils.MAXVALLEN:
+                spec['full' + key] = spec[key]
+                spec[key] = hashlib.sha1(spec[key].encode()).hexdigest()
+
+    @staticmethod
+    def _fix_infos_size(spec):
+        # We enforce a utils.MAXVALLEN // 10 size limits for subkey values in
+        # infos; this is because MongoDB cannot index values longer than 1024
+        # bytes.
+        for field in list(spec.get("infos", {})):
+            value = spec["infos"]
+            if len(value) > utils.MAXVALLEN:
```

Comment:
That does not seem consistent with the above comment (`utils.MAXVALLEN // 10`?).

Reply:
BTW, is there a reason to have two distinct methods here?

Reply:
Same reason I had two commits, uncleaned dev process...

Reply:
Wrong fix now fixed.